src/amplitude/http_client.py

```diff
@@ -134,4 +134,14 @@ def post(url: str, payload: bytes, header=None) -> Response:
         except timeout:
             result.code = 408
             result.status = HttpStatus.TIMEOUT
+        except error.HTTPError as e:
+            try:
+                result.parse(e)
+            except:
+                result = Response()
```

Comment:
result.parse was designed to parse Amplitude backend server response, may not work taken HTTPError as input. Set code, status and body directly here is recommended.

Reply:
I understand that, that's why it's wrapped in `try ... except` that handles that case.